README.md

```diff
@@ -0,0 +1,275 @@
+servers, web frameworks, and web application into a single method call.
+
+The exact details of this are described in the [Rack Specification],
+which all Rack applications should conform to.
+
+## Installation
+
+Add the rack gem to your application bundle, or follow the instructions provided by a [supported web framework](#supported-web-frameworks):
+
+```bash
+$ bundle add rack
+```
+
+## Usage
+
+Create a file called `config.ru` with the following contents:
+
+```ruby
+run do |env|
+  [200, {}, ["Hello World"]]
+end
+```
+
+Run this using the rackup gem or another [supported web server](#supported-web-servers).
+
+```bash
+$ gem install rackup
+$ rackup
+$ curl http://localhost:9292
+Hello World
+```
+
+## Supported web servers
+
+Rack is supported by a wide range of servers, including:
+
+* [Agoo](https://github.com/ohler55/agoo)
+* [Falcon](https://github.com/socketry/falcon)
+* [Iodine](https://github.com/boazsegev/iodine)
+* [NGINX Unit](https://unit.nginx.org/)
+* [Phusion Passenger](https://www.phusionpassenger.com/) (which is mod_rack for Apache and for nginx)
+* [Puma](https://puma.io/)
+* [Thin](https://rubygems.org/gems/thin)
+* [Unicorn](https://yhbt.net/unicorn/)
+* [uWSGI](https://uwsgi-docs.readthedocs.io/en/latest/)
+* [Lamby](https://lamby.custominktech.com) (for AWS Lambda)
+
+You will need to consult the server documentation to find out what features and
+limitations they may have. In general, any valid Rack app will run the same on
+all these handlers, without changing anything.
+
+### Rackup
+
+Rack provides a separate gem, [rackup](https://github.com/rack/rackup) which is a generic interface for running a Rack application on supported servers, which include `WEBRick`, `Puma`, `Falcon` and others.
+
+## Supported web frameworks
+
+These frameworks and many others support the [Rack Specification]:
+
+* [Camping](http://www.ruby-camping.com/)
+* [Coset](https://leahneukirchen.org/repos/coset/)
+* [Hanami](https://hanamirb.org/)
+* [Padrino](https://padrinorb.com/)
+* [Ramaze](http://ramaze.net/)
+* [Roda](https://github.com/jeremyevans/roda)
+* [Ruby on Rails](https://rubyonrails.org/)
+* [Rum](https://github.com/leahneukirchen/rum)
+* [Sinatra](https://sinatrarb.com/)
+* [Utopia](https://github.com/socketry/utopia)
+* [WABuR](https://github.com/ohler55/wabur)
+
+## Available middleware shipped with Rack
+
+Between the server and the framework, Rack can be customized to your
+applications needs using middleware. Rack itself ships with the following
+middleware:
+
+* `Rack::CommonLogger` for creating Apache-style logfiles.
+* `Rack::ConditionalGet` for returning [Not Modified](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/304) responses when the response
+  has not changed.
+* `Rack::Config` for modifying the environment before processing the request.
+* `Rack::ContentLength` for setting content-length header based on body size.
+* `Rack::ContentType` for setting default content-type header for responses.
```

Comment:
```suggestion
* `Rack::ContentLength` for setting a Content-Length header based on body size.
* `Rack::ContentType` for setting a default Content-Type header for responses.
```

I added articles to the sentence, perhaps this makes it flow easier? (Not a native speaker.)

"Header-casing" the mention of specific headers may make them easier to disambiguate from the rest of the text.

Reply:
Rack 3 requires lower case headers, so it will be confusing, perhaps use backticks?

Reply:
Wise choice, backticks.

Reply:
```suggestion
* `Rack::ContentLength` for setting a `content-length` header based on body size.
* `Rack::ContentType` for setting a default `content-type` header for responses.
```